test/langtools/tools/javac/reflect/DenotableTypesTest.java

```diff
@@ -146,4 +146,46 @@ static <X extends Object & Runnable> void test7() {
         X x = null;
         consume(x);
     }
+
+    interface Adder<X> {
+        void add(Adder<X> adder);
+    }
+
+    @CodeReflection
+    @IR("""
+            func @"test8" (%0 : java.util.List)void -> {
+                  %1 : Var<java.util.List> = var %0 @"list";
+                  %2 : java.util.List = var.load %1;
+                  %3 : int = constant @"0";
+                  %4 : DenotableTypesTest$Adder = invoke %2 %3 @"java.util.List::get(int)java.lang.Object";
+                  %5 : java.util.List = var.load %1;
+                  %6 : int = constant @"1";
+                  %7 : DenotableTypesTest$Adder = invoke %5 %6 @"java.util.List::get(int)java.lang.Object";
+                  invoke %4 %7 @"DenotableTypesTest$Adder::add(DenotableTypesTest$Adder)void";
+                  return;
+            };
+            """)
+    static void test8(List<? extends Adder<Integer>> list) {
+        list.get(0).add(list.get(1));
+    }
+
+    static class Box<X> {
+        X x;
+    }
+
+    @CodeReflection
+    @IR("""
+            func @"test9" (%0 : java.util.List)void -> {
+                  %1 : Var<java.util.List> = var %0 @"list";
+                  %2 : java.util.List = var.load %1;
+                  %3 : int = constant @"0";
+                  %4 : DenotableTypesTest$Box = invoke %2 %3 @"java.util.List::get(int)java.lang.Object";
+                  %5 : java.lang.Integer = field.load %4 @"DenotableTypesTest$Box::x()java.lang.Object";
+                  %6 : Var<java.lang.Integer> = var %5 @"i";
+                  return;
+            };
+            """)
+    static void test9(List<? extends Box<Integer>> list) {
```

Comment:
Strictly speaking this test isn't needed, as it passed even w/o the fix (as cloning a VariableSymbol doesn't incur in any well-formedness check. This means a new var symbol is created with a captured variable owner, but the owner is then "normalized" to its erasure by `ReflectMethods::typeToTypeElement`, so the crash is avoided. Anyway, I preferred to add the correct call to `erasure` even for fields, and add a test, to make sure we don't regress.